### DISCUSSION TITLE
How to control the grain orientation?

### DISCUSSION THREAD

**Discussion by hubocqjtu** [neperfepx/neper#274] [Q&A] [answered]
Dear all,

I am confused that the -ori. How to control grain orientation using -T.

i.e., How to make the direction of the grains 0,45 or 90 degrees to a certain axis (x,y or z axis) ? Don't use -domain rotation, just use -ori or related commands?
I tried this code but failed.

$ neper -T -n 100 -morpho "voronoi,aspratio(2,2,0.5)" -ori "fiber(0,0,1,0,0,1)"

![20210722160733](https://user-images.githubusercontent.com/87802682/126608617-d08890be-c8e3-48b3-88f3-4bdae5edf917.jpg)

Thanks!

Bolton

**Comment by rquey** [comment, marked as answer]
`-fiber(0,0,1,0,0,1)` gives you all 001 crystal axes parallel to the Z sample axis (and random orientations about the fiber axis).

Your orientation distribution cannot be generated by Neper. So, you have to generate the orientations yourself and load them using `-ori file(...)`.

**Reply by hubocqjtu** [reply]
Thank you so much! I will try.
